### ISSUE TITLE
Able to login using normal login form with AAD SSO configured

### ISSUE THREAD

**Issue by samotelf** [grokability/snipe-it#11667]
### Debug mode

- [X] I have enabled debug mode
- [X] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

### Describe the bug

I've managed to configure AAD SSO with Snipe-IT, and when in the browser I insert the snipe-it url the AAD SSO screen appears as it should. So far, so good, however, I think it's strange that when I logout (from the user top right menu) I'm able to login using the normal Login form, even with the option Make SAML the primary login (You can use '/login?nosaml' to get to the normal login page.) checked.
I don't think this is normal behavior. 
As per documentation it says:

"SAML Force Login
When this checkbox is enabled, you will not see a login form of Snipe-IT anymore when you go to the Snipe-IT website. Instead it will redirect you directly to the IdP SAML Login." and this is true, but imho it should also be true when you logout.

Any insights on this?

Thank you very much.

### Reproduction steps

1. Login in Snipe-IT using the Microsoft SSO (since it's configured with Azure).
2. Logout from the user option in the top right corner.
3. Being able to login using the normal Login Form.
...

### Expected behavior

Not being able to login using the normal Login Form.
In my opinion, the user shouldnt be able to login using the normal Login Form, or at least if he tries to do so, get an error/link to login via SSO again.

### Snipe-IT Version

6.0.9

### Operating System

CentOS 7

### Web Server

Apache

### PHP Version

7.4.30

### Operating System

Windows

### Browser

Chrome

### Version

101.0.4951.64

**Comment by samotelf** [comment]
Update:
If in the .env file there's the line "REQUIRE_SAML=true" I can only in fact login only with SAML, but unfortunately using https://assets.example.com/login?nosaml (this is useful in case SSO is down) won't let me login using the normal login form.

**Comment by snipe** [comment]
Well yeah, that's the point tho. `REQUIRE_SAML=true` literally disables any ability to login using the regular web UI, even with the `/login?nosaml` parameter.

**Comment by samotelf** [comment]
> Well yeah, that's the point tho. `REQUIRE_SAML=true` literally disables any ability to login using the regular web UI, even with the `/login?nosaml` parameter.

Hello. 
Didn't know that. I was just going by trial and error, I'll take that line out and wait for feedback. 
Thank you.

**Comment by uberbrady** [comment]
Yeah, we actually allow our hosted customers do have that setting enabled, but we discourage it because it's hard for us to log in with our own user to look at their instance when they need us to.

Regardless, if it turns out that things are working as expected and your users are happy, please do close the issue once if they let you know that the system is working OK. Thanks!

**Comment by snipe** [comment]
Yeah, for hosted customers, if we enable that flag, we explain that our ability to provide support will be limited unless they want to create a user for us in their SAML provider (which a few have done.)

**Comment by samotelf** [comment]
Ok, I really don’t know what to say.
If it’s supposed for the user login using the login form after logout when SAML is configured, I guess you could close the ticket.
:) thank you all.

**Comment by chais0n** [comment]
I still think this is an issue. 

When I enabled 'Make SAML the primary login' I'm always presented with the normal username/password login form with the little link undereath to login with SAML. 

I would expect that I don't get the username/password login box at all.

**Comment by uberbrady** [comment]
Can you try to set `REQUIRE_SAML` to `true` in the `.env` file and let us know if that does what you're expecting? That *should* force SAML *only* - and not allow 'local' logins at all. Can you confirm?

**Comment by snipe** [comment]
Hi there - We haven't heard back in a bit, so I'm going to close this ticket for now, but will re-open it if you're still having issues.

![hackerman-hacking](https://user-images.githubusercontent.com/197404/31936003-27a7f6d8-b865-11e7-8f5e-eb03062925f4.gif)

**Comment by seal61** [comment]
> Can you try to set `REQUIRE_SAML` to `true` in the `.env` file and let us know if that does what you're expecting? That _should_ force SAML _only_ - and not allow 'local' logins at all. Can you confirm?

Even if this is set in the .env after logout the login form with the small link "Login via SAML" is present AND local accounts without SAML can be used. Using latest `snipe-it v8.3.2 build 19905 (g028b4e7b7)`

What should the setting `Admin Settings > Update Security Settings > Custom logout URL` look like?

**Comment by seal61** [comment]
@uberbrady any ideas? :/

**Comment by uberbrady** [comment]
Huh - weird. I just enabled that on our own instance (`REQUIRE_SAML` set to `true` in the `.env`) and when I logout, I get this screen:

<img width="418" height="293" alt="Image" src="https://github.com/user-attachments/assets/98a9e3b7-985e-44d8-9c35-6f085f2b792a" />

Is it possible that you maybe have configuration caching? Try `php artisan config:clear` - any different behavior?

**Comment by seal61** [comment]
still getting forwarded to this login page: (on /login)

<img width="392" height="487" alt="Image" src="https://github.com/user-attachments/assets/e73ea631-d9b6-44d1-8cec-80874479f1b1" />

performed the cache clear successfully with `docker exec -it snipe-it php artisan config:clear` and also recreated the containers. My `.env` looks like:

```
APP_ENV=production
APP_DEBUG=false
APP_KEY='REDACTED'
APP_URL='REDACTED'
APP_TIMEZONE='Europe/Berlin'
APP_LOCALE='en-US'
MAX_RESULTS=500
APP_ALLOW_INSECURE_HOSTS=true

## UPLOADED FILE STORAGE SETTINGS
PRIVATE_FILESYSTEM_DISK=local
PUBLIC_FILESYSTEM_DISK=local_public

## DATABSE SETTINGS
REDACTED


## DATA PROTECTION
ALLOW_BACKUP_DELETE=false
ALLOW_DATA_PURGE=false

## IMAGE LIBRARY
IMAGE_LIB=gd

## SECURITY HEADER SETTINGS
APP_TRUSTED_PROXIES=REDACTED

## MISC
APP_FORCE_TLS=true

## FORCE SAML
RQUIRE_SAML=true
```

Could you maybe share your configuration in snipe-it? Are you also using te docker image? I'm on `snipe/snipe-it:v8.3.4`

... EDIT: writing this i found the typo... "RQUIRE_SAML" is probably not right... *sigh*...

**Comment by snipe** [comment]
> ... EDIT: writing this i found the typo... "RQUIRE_SAML" is probably not right... sigh...

![Image](https://github.com/user-attachments/assets/2ef73b6f-d711-4483-ac63-23ce70e2f452)

I'm so sorry, I couldn't resist 😆
